### PR TITLE
Zubing add GoogleDoc&Dropbox links on the profile

### PR DESCRIPTION
**Pull request by bienzguoa**
# Description
PRIORITY (URGENT) Jae: Add “Admin Document” and “Link to Media Files” to Profile page (WIP Zubing)
When creating a new user (Other Links → User Management → Create New User), the options exist to add an “Admin Document” and a “Link to Media Files”. 
If you add them during setup, they get added there. However, if you forget or they change, there’s no easy way to edit them. I’d like a permanent link above to the easy place to edit these because every person has them. 

## Related PRS (if any):
This frontend PR is related to frontend PR #1001 :

1. It fixed "add" and "update" button for the links.

## Main changes explained:
- Update UserLinks.jsx for including Dropbox link
- Update EditLinkModal.jsx for making names(Google Doc and Dropbox Folder) fixed and links fetched ti corresponding place.
- Update weeklySummaries.js for updating links to Weekly Summaries Reports page.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user
4. go to Other Links→ User Management→ Create New User
5. create users: add “Admin Document” and “Link to Media Files” / add only one of them/ add none of them
6. go to these users' Profile→ Links Edit
7. verify:
there are name of Google Doc and Dropbox Folder fixed and corresponding box to add links.
for users created, if links were added when creating, check if the links are in the box and the names show on the profile under "Link Edit".
for users created, if links were not added when creating, check if the boxes are empty and there is no name show on the profile under "Link Edit".
delete the links and click" update, save changes" , the names on the profile under "Link Edit" would disappear.
Testing video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/89e7778d-d176-4e70-a710-8b6afa61030c

8. go to Reports→ Weekly Summaries Reports
9. Verify for Google Doc link:
the added link are updated to the Weekly Summary Reports page.
if modify the link in profile, this link in the Weekly Summary Reports page are changed correspondingly.
if delete the link, it shows "not present" in the Weekly Summary Reports page when clicking the Google Icon.
Testing video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/3d2eb9fb-0fa1-40fc-a91b-e9424fe36360

10. Verify for Media Folder(Dropbox link):
the added link are updated to the link box when submitting weekly summary.
if modify the link in profile, this link in the box when submitting weekly summary are changed correspondingly.
after submitting the summary, the link shows in the Weekly Summary Reports correctly.
Testing video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/193f92e4-1036-401e-b474-af65c0ef147d

### CODE REVIEW

**Comment by beblicarl** [comment]
I tested both scenarios and can confirm that this functionality works as intended

issue #980-1

issue 980-2

**Review by beblicarl — DISMISSED** [review]

**Review by yaow62023 — DISMISSED** [review]
Hi, @bienzguoa. I have tested your PR, both features are work as intended. The edit link button can display the link and the nam of the link. Also, have place to add other links. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/56cffd91-2e58-417f-9535-883ef96c06c7

**Review by HectorAgudelo — COMMENTED** [review]
Placement of links seems to work as expected. however, I observed that if the links are modified, following the conditions in the code, the new links do not persist when the user comes back to the edit window after being somewhere else in the app. So, i was wondering if the links placed  when creating users are editable.
PR#980-onelink

PR#980-nolink
PR#980-2links

**Comment by Jiangwei-shi** [comment]
Hi, bienzguoa. I have tested your PR, it works as intended. The edit link button can display the link and can replace the link.
image

**Review by siddharthmohite — DISMISSED** [review]
This worked as intended. On creating a new user with their Dropbox and google drive, links are editable in the user profile.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/37925823/e576a16f-a01e-42c6-8395-11a7d2c8265b

**Review by ramyaram2092 — COMMENTED** [review]
Hello @bienzguoa . I tested the PR locally. 

**New profile :** 
I see the drive and drop box link are getting successfully added to profile if its added at new user creation

**Existing profile  :** 
For the existing profile I click on edit link and update the google drive and drop box link. 
If I refresh the page i dont see the changes. I dont think it is getting saved for the existing users

Could you confirm this behaviour  and let me know if it is expected

**Review by shiva2096 — CHANGES_REQUESTED** [review]
Hi @bienzguoa 

I found an issue.
The updates made for an existing user are not getting saved.
Please review this video:

Steps to recreate:
1) I created a user and added the google docs link only.
2) Next I went to user page and tried to update the drop box link.
3) After clicking update, the drop box link is not visible on profile page.
4) I revisited again by refreshing the page, still not visible.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26687177/ef79fc7d-21b7-46ee-a717-46ae4640b227

Please let me know if you have any questions.

**Review by xaanders — CHANGES_REQUESTED** [review]
Hi! Tested you PR. You've done a good job. 

- Registered user with links and they are in the profile both on page and inside edit form:

with

- registered without links and there were no links in profile:
without

I have also tried with only one link added to a user, and it works as intended.

But only Google Doc link in the profile lead to a broken path: 
link

**Comment by bienzguoa** [comment]
> Hi @bienzguoa
> 
> I found an issue. The updates made for an existing user are not getting saved. Please review this video:
> 
> Steps to recreate:
> 
> 1. I created a user and added the google docs link only.
> 2. Next I went to user page and tried to update the drop box link.
> 3. After clicking update, the drop box link is not visible on profile page.
> 4. I revisited again by refreshing the page, still not visible.
> 
>  Screen.Recording.2023-07-08.at.9.16.20.PM.mov 
> Please let me know if you have any questions.

Hi [shiva2096](https://github.com/shiva2096),
Thanks for your review! I think it's because the "update" button for changing data is not working, it should be in the bug list, but not directly relevant to this PR.

**Comment by bienzguoa** [comment]
> Hello @bienzguoa . I tested the PR locally.
> 
> **New profile :** I see the drive and drop box link are getting successfully added to profile if its added at profile creation
> 
> **Existing profile :** For the existing profile I click on edit link and update the google drive and drop box link. If I refresh the page i dont see the changes. I dont think it is getting saved for the existing users
> 
> Could you confirm this behaviour and let me know if it is expected

Hi [ramyaram2092](https://github.com/ramyaram2092),
Thanks for your review! Yes, now the "update" button is not working so the edited links can't be kept, it should be in the bug list, but not directly relevant to this PR.

**Comment by bienzguoa** [comment]
> Hi! Tested you PR. You've done a good job.
> 
> * Registered user with links and they are in the profile both on page and inside edit form:
> 
> with
> * registered without links and there were no links in profile:
> 
> without
> I have also tried with only one link added to a user, and it works as intended.
> 
> Updating links from profile functionality still need to be made.
> 
> But only Google Doc link in the profile leads to a broken path: link

Hi [xaanders](https://github.com/xaanders),
Thank you! The update function need to be solved in another PR, for the broken path you mentioned, I tested again and didn't encounter such problem, could you check the Google doc link and try again?

**Review by xaanders — APPROVED** [review]
I'm sorry for misleading. I'd been using google drive link, so the path got broken. 
Works as intended!

**Review by tdkent — APPROVED** [review]
Local testing confirms

- when creating a user as an admin, any links added to the `Admin Document` and `Link to Media Files` inputs are added to the new user's account.
- The **Edit Links** form now includes active Google and Dropbox input fields.
- the current links, if any, are correctly displayed in the **Edit Links** form.
- if no links were added when the user was created, the corresponding **Edit Form** fields are blank.

980-one-link

Issues encountered (potentially outside scope of the PR):

- any updates made in the **Edit Links** form are lost when the page is reloaded.
- clicking any of the `+` buttons in **Edit Links** causes the site to crash (see below).

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/b0d97459-3de3-4b9c-864f-89e5be582c80

Testing Video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/6defa5ab-7ecc-43dd-847e-b7d5eed4818b

**Comment by one-community** [comment]
> 4\. I revisited again by refreshing the page, still not visible.

The link should also update to the Weekly Summaries Reports page. Please confirm saving it here also adds it there. This is very important.

**Comment by one-community** [comment]
"Dropbox Folder" should also say "Media Folder"

"enter" should be "Enter"

image

**Comment by one-community** [comment]
This is where these links should also be connecting to and confirmed as updating if changes are made on the Profile Page: 
image

**Review by BaileyM96 — CHANGES_REQUESTED** [review]
Tested the new changes and everything works great when you add the links to the profile upon creation. But I did run into an issue.

## Issues

As you already know based on other feedback in the PR you cannot update the links once the user has been created.

Steps to reproduce the problem:
1.Navigate to the created user
2.View their profile
3.Click on "Edit"
4.Add a new link to the field and update

As you see the links never update on the page even if you refresh the page.

The next Issue I had is in regards to the UI for the update button. The update button should be changed to "Save Changes" not "update" since the button on the profile page says "Save Changes" Here is a few images:

Screen Shot 2023-07-10 at 3 42 56 PM

Screen Shot 2023-07-10 at 3 44 25 PM

This is a small issue but this will help the website stay uniformed.

## Video Walkthrough

[Link updates.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108101478/133b58d6-da64-4357-b44d-9ac6aeb11f3a)

**Review by luisarevalo21 — COMMENTED** [review]
Hello [bienzguoa](https://github.com/bienzguoa),

I've tested your branch and works as intended. Creating a new user and adding their dropbox and google doc, is added to their profile so others can view. 

As others have noted, attempting to add these links to existing accounts does not work. 
Also, clicking the "+" next to the input field breaks the application. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/673eb414-780e-44c4-a1c5-36e720f5e5da

**Comment by ramyaram2092** [comment]
@one-community  This PR has  5 approvals

**Comment by one-community** [comment]
> @one-community This PR has 5 approvals

Thank you. Requested changes need to be addressed, and branch conflicts. Then I'll move it to final review.

**Comment by bienzguoa** [comment]
> > 4. I revisited again by refreshing the page, still not visible.
> 
> The link should also update to the Weekly Summaries Reports page. Please confirm saving it here also adds it there. This is very important.

Hi Jae, 
For the google doc link, the code for updating it to the Weekly Summaries Reports page has been done, but it cannot be tested and confirmed until the "update" button for saving edited links is fixed in another task(I've talked with Yubo about that). For the media folder, it's unique for every week( every week has a different folder), so the data for this link is updating from the link we filled in when submitting the weekly summary.
Screenshot 2023-07-13 at 1 58 05 PM

Screenshot 2023-07-13 at 1 56 50 PM

**Review by iTvX — APPROVED** [review]
I have tested your code, it works fine. In the first time, I add only one of them, it works. In the second time, I add both link, it works too.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/b80ec445-1a1e-452c-84bb-4c400febf228

File: src/components/UserProfile/UserLinks/UserLinks.jsx
Comment (iTvX) [inline src/components/UserProfile/UserLinks/UserLinks.jsx:13]: The key attribute of the <a> tag is using item.link, but there is no link property in the item object.  I think item.Link should be used as the value for the key attribute.

**Comment by one-community** [comment]
> it's unique for every week( every week has a different folder), so the data for this link is updating from the link we filled in when submitting the weekly summary.

Correct. People shouldn't be editing this folder every week though. It's now input when I set someone up as a new user and a new warning will pop up if they try and edit it.

**Comment by AnishPandita** [comment]
Hi @bienzguoan the functionality works well. I tried with one link as well as both.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/38046cdd-0264-467c-929b-dc61d6b9ca6c

**Review by DavidC0126 — APPROVED** [review]
Hi @bienzguoa, I tested your PR and everything works perfectly! I create a new user and add two links at the same time. When I clicked the links I can be directed to the right URL.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/06b46489-a797-41ab-8bf4-40130398be3a

**Comment by Shreyj14** [comment]
Hi @bienzguoa 

When a new user is created, at that point if you add the user google doc link and dropbox link it works as intended.
Screenshot 2023-07-20 at 2 21 37 PM
Screenshot 2023-07-20 at 2 21 59 PM

But once the profile is already created and you try to add the dropbox link, it does not show up in the profile. Can you please recheck if that is expected behavior?
Screenshot 2023-07-20 at 2 22 38 PM

**Review by Shreyj14 — CHANGES_REQUESTED** [review]
The links are not displayed on the profile if a google-doc/dropbox link is added for an already exisiting user.

File: src/components/UserProfile/UserLinks/UserLinks.jsx
Comment (GKD890) [inline src/components/UserProfile/UserLinks/UserLinks.jsx:21]: During the test I found if an added link doesn't start with "http", it redirects to internal page. But the link I've added is `www.google.com`. I'm just wondering if that's related to this PR?

**Review by GKD890 — APPROVED** [review]
It seems work as intended. The media folder will be changed after submitting a weekly summary. The changed link will be reflected in 'Edit' link in User Profile page.

File: src/components/UserProfile/UserLinks/UserLinks.jsx
Comment (bienzguoa) [reply src/components/UserProfile/UserLinks/UserLinks.jsx:21]: It's not related to this PR(I think this part of code was modified before, so i updated it in this way to resolve the conflicts). In UserLinks.jsx, it's set that the links need to include 'http', otherwise it will redirect to internal page.

**Review by Indujaks — DISMISSED** [review]
Tested the changes by creating a new user account, and the media folder redirects to dropbox, works as expected!
Screenshot 2023-07-27 at 4 47 06 PM

**Review by winghojackyli — DISMISSED** [review]
Hi @bienzguoa , I've tested your PR and it works as expected. 

### Create users with both google doc and media folder, only media and neither of them
I can edit and delete those links as well.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/4546b054-b1c0-4d40-a301-63dcd08521e0


### Weekly Summary Report Page
The links are reflected on the weekly summary report page. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/991942e7-b2da-4aa6-95cf-96cfc7611e58


### Submit Weekly Summary Report
After submitting the report, the media folder link works at the report page. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/985140d6-33d8-4200-8b87-8428c985e3b1

**Review by GKD890 — DISMISSED** [review]
approve again after resolving conflicts.

**Review by Shreyj14 — APPROVED** [review]

**Review by GKD890 — APPROVED** [review]

**Review by BaileyM96 — DISMISSED** [review]
Good afternoon @bienzguoa I tested your new changes after my requested changes. Everything works fine I am able to add admin documents and media files to a user when I create their profile, I can then edit the new links and remove them and save the changes.

I also tested existing accounts to see if I can edit and remove their admin links and that works too, great job.

## Video Walkthrough

[PR #980 add GoogleDoc&Dropbox links in profile_.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108101478/0dc92d75-8c60-4500-9924-f1dcc28c9ff4)

**Comment by Lin1404** [comment]
Hi @bienzguoa, I tested your PR and it works perfectly! 
image

**Review by Lin1404 — DISMISSED** [review]

**Comment by beblicarl** [comment]
I tested your new changes and everything works

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.08.07-10_59_13.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/217e9bcc-df49-4fef-b64d-7a27991dce50)

**Review by shiva2096 — APPROVED** [review]
Hi @bienzguoa 

I re-tested the PR, and everything looks great.

**Review by veronicacheng2 — APPROVED** [review]
Hi @bienzguoa, I have tested the PR and everything works as intended

### Test 1 (Can successfully edit the link in user profile page even the link wasn't filled in during user creation):
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/0ab686ae-2ca6-4c0c-aa37-8e2a0bce9e6e

### Test 2 (The edited link is shown in weekly summaries report)


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/0fd2da65-a714-40f7-b5dd-042f861cabab


### Test 3 (The newly added dropbox link will show in weekly summaries)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/e34e3567-80f4-4920-b9f3-bb43c87e1b27

**Comment by iTvX** [comment]
I have followed your all instruction, no bugs found. Everything looks fine.
Test 1:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/08380c63-52f6-46e5-9385-94354aa85794

Test 2:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/494a2519-8ff6-41cc-a3b8-28441cfcb382

Test 3:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/e434ffee-fcb7-4f05-9d18-8e430960e8e0

**Review by iTvX — APPROVED** [review]
I have followed your all instruction, no bugs found. Everything looks fine.
Test 1:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/08380c63-52f6-46e5-9385-94354aa85794

Test 2:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/494a2519-8ff6-41cc-a3b8-28441cfcb382

Test 3:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/e434ffee-fcb7-4f05-9d18-8e430960e8e0

**Comment by iTvX** [comment]
I approve your request, great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/3e11297b-8ced-466a-be8d-2edc466009f8


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/c14a575b-0552-497c-8d70-8100e8200cb9

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/3b9dd75c-9622-41c1-b616-6d1e3fe4dd85

**Review by ThapeloMasasa — APPROVED** [review]
The PR works as intended for the functionality of the media and google docs link
Screenshot 2023-08-11 at 7 03 09 PM
Screenshot 2023-08-11 at 7 03 22 PM

**Review by yunshan111 — APPROVED** [review]
Tested your PR, and everything works well.

**Review by Abiddy — APPROVED** [review]
Works nicely!

[Screencast from 08-12-2023 08:29:44 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/77697776/0440ac6a-87da-4f29-b667-567de5473ebf)

**Comment by bienzguoa** [comment]
@one-community Here are 6 approvals.

**Comment by one-community** [comment]
Great!! Moving this to final review!

**Pull request merged**
